src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,200 @@
+package chatbot;
+import java.time.LocalDate;
+// import java.time.format.DateTimeFormatter;
+// import java.time.temporal.ChronoUnit;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+/**
+* Parser class for the bot, to handle incoming inputs from the user.
+*/
+public class Parser {
+    private static final String UNRECOGNIZED_COMMAND = "Sorry Sir, I do not understand that command.";
+    private static final String NOT_VALID_NUMBER = "Please enter a valid number Sir.";
+    private static final String COMMAND_REQUIRES_NUMBER = "Sorry Sir, this command requires a number.";
+    private static final String MISSING_TASK_INFO = "Sorry Sir, the <%s> command cannot be empty.";
+
+    /**
+    * Displays the farewell message to the user.
+    */
+    public static void displayFarewell() {
+        String byeMessage = "Farewell Sir. May you have a wonderful day.";
+        Ui.displayMessage(byeMessage);
+    }
+    /**
+    * Returns the boolean of whether the bot should expect more inputs from the user,
+    * after processing the inputText String from the user.
+    *
+    * @param    inputText    input of the user
+    * @param    taskList     TaskList of the user
+    * @param    storage      Storage of the user denoting the location of the save file
+    * @return                the boolean to denote whether there are more future inputs
+    * @see      TaskList
+    * @see      Storage
+    */
+    public static boolean parseText(
+            String inputText, TaskList taskList, Storage storage) { // returns true if bot should continue parsing text
+        inputText = inputText.trim();
+        String[] inputStringArray = inputText.split(" ");
+        try {
```

Comment:
Might be a bit cleaner if you create some specific exceptions and you handle all the exceptions in a single try-catch? Probably not arrowheading right now but might become a risk if you add on more functionality per case